apps/files/lib/BackgroundJob/ScanFiles.php

```diff
@@ -87,6 +94,22 @@ protected function runScanner(IUser $user) {
 		\OC_Util::tearDownFS();
 	}
 
+	/**
+	 * Find all storages which have unindexed files and return a user for each
+	 *
+	 * @return string[]
+	 */
+	private function getUsersToScan(): array {
+		$query = $this->connection->getQueryBuilder();
+		$query->select($query->func()->max('user_id'))
+			->from('filecache', 'f')
+			->innerJoin('f', 'mounts', 'm', $query->expr()->eq('storage_id', 'storage'))
+			->where($query->expr()->lt('size', $query->createNamedParameter(0, IQueryBuilder::PARAM_INT)))
+			->groupBy('storage_id');
```

Comment:
```suggestion
			->groupBy('storage_id')
			->setMaxResults(self::USERS_PER_SESSION);
```

better save than sorry?

Reply:
yep sounds sane